src/Controller/CompanyController.php

```diff
@@ -4,73 +4,129 @@
 
 namespace App\Controller;
 
+use App\Entity\Company;
+use App\Repository\CompanyRepository;
+use Symfony\Component\HttpFoundation\JsonResponse;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
-use Symfony\Component\Form\Extension\Core\Type\TextType;
-use Symfony\Component\Form\Extension\Core\Type\SubmitType;
-use Psr\Log\LoggerInterface;
 
 /**
  * new class for company controller
  */
 class CompanyController extends AbstractController
 {
 
-    public $form;
-    public $name;
-    public $description;
-    public $data;
-    private $logger;
+    private CompanyRepository $companyRepository;
+    /**
+     * __construct
+     *
+     * @param  mixed $companyRepository
+     * @return void
+     */
+    public function __construct(CompanyRepository $companyRepository)
+    {
+        $this->companyRepository = $companyRepository;
+    }
     
-    public function __construct(LoggerInterface $logger)
+    public function addCompany(Request $request): Response
     {
-        $this->logger=$logger;
+        $company = new Company();
+        $company->setName($request->get('name'));
+        $companySaved = $this->companyRepository->save($company);
+        return new JsonResponse(
+            [
+                'Saved new company' => $companySaved
+            ]
+        );
     }
     
     /**
-     * loadTemplate
+     * get all companies from the database
      *
+     * @param  mixed $doctrine
      * @return Response
      */
-    public function loadTemplate(): Response
+    public function readCompany(): Response
     {
-        return $this->render('company/index.html.twig');
-    }
 
-     /**
-     * loadTemplate
+        $rows = $this->companyRepository->select();
+        return new JsonResponse([
+            $rows
+        ]);
+    }
+    /**
+     * gets the company by the ID
      *
-     * @param  mixed $request
+     * @param  mixed $id
      * @return Response
      */
-    public function addCompany(Request $request): Response
+    public function readCompanyByID(int $id): Response
     {
-        $this->form = $this->createFormBuilder()
-            ->add('name', TextType::class)
-            ->add('description', TextType::class)
-            ->add('Add', SubmitType::class)
-            ->getForm();
-   
-        $this->form->handleRequest($request);
 
```

Comment:
Remove empty line. After the funtion declaration there must be no empty line before first line from function body. Check for all functions in file.